ctk/common/src/main/kotlin/org/codice/compliance/utils/RestAssuredExtensions.kt

```diff
@@ -25,20 +26,38 @@ import org.codice.compliance.verification.binding.PostBindingVerifier
 import org.codice.compliance.verification.binding.RedirectBindingVerifier
 import org.codice.security.saml.SamlProtocol
 
+/** Response extension functions **/
+private const val BINDING_UNSUPPORTED_MESSAGE = "Binding is not currently supported."
+
 fun Response.determineBinding(): SamlProtocol.Binding {
     return with(this) {
         when {
             isPostBinding() -> SamlProtocol.Binding.HTTP_POST
             isRedirectBinding() -> SamlProtocol.Binding.HTTP_REDIRECT
-            else -> throw UnsupportedOperationException("Binding is not currently supported.")
+            else -> throw UnsupportedOperationException(BINDING_UNSUPPORTED_MESSAGE)
         }
     }
 }
+
 fun Response.getBindingVerifier(): BindingVerifier {
     return when (this.determineBinding()) {
         SamlProtocol.Binding.HTTP_REDIRECT -> RedirectBindingVerifier(this)
         SamlProtocol.Binding.HTTP_POST -> PostBindingVerifier(this)
-        else -> throw UnsupportedOperationException("Binding is not currently supported.")
+        else -> throw UnsupportedOperationException(BINDING_UNSUPPORTED_MESSAGE)
+    }
+}
+
+fun Response.getLocation(): String? {
+    return when (this.determineBinding()) {
+        SamlProtocol.Binding.HTTP_REDIRECT -> {
+            // Extracted according to Binding 3.4.4
+            this.getHeader("Location")
+        }
+        SamlProtocol.Binding.HTTP_POST -> {
+            // Extracted according to Bindings 3.5.4
+            this.extractSamlResponseForm()?.getAttribute(ACTION)
+        }
+        else -> throw UnsupportedOperationException(BINDING_UNSUPPORTED_MESSAGE)
```

Comment:
❓ I wonder if it might be clearer/better if we added a specific case for artifact binding even though it would just do the same as the else branch.

Reply:
I see this extension function being used for any protocol which could use any binding, so I don't think we need to have an explicit check for artifact.